spec/views/application_templates/edit_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+include RSpecHtmlMatchers
+
+describe 'application_templates/edit.haml' do
+  before :each do
+    @template = create :application_template
+    @top_question = create :question, application_template: @template
+    @middle_question = create :question, application_template: @template
+    @bottom_question = create :question, application_template: @template
+  end
+  it 'includes a link to view application' do
+    render
+    action_path = application_template_path @template
+    expect(rendered).to have_tag 'a', with: { href: action_path }
+  end
+  it 'contains a form with questions' do
```

Comment:
I'd like to combine this and the following few tests about it into one test about that form. That way, we can specifically say that those inputs are part of this form, and that we expect a form which matches that precise description.